Ability to silence DM messages from HeyFireball #14

JacobJerrell opened the pull request:
PM Functionality is fully functional for InMemoryStorage. Testing is needed for Azure, and Redis. Some housekeeping was done; commented out unused functions/methods, corrected formatting.

The bot accepts the following commands:
@[botname] setpm  -- Acts as a toggle
@[botname] setpm on
@[botname] setpm off

JacobJerrell committed:
developing under develop now

switched to dev branch, back to InMemoryStorage only

JacobJerrell committed:
housekeeping

Removed debugging lines, got rid of/commented out unused code

JacobJerrell committed:
formatting corrections. pm preference in azure and InMemoryStorage

Remove spaces. PM Preference options are now in Azure, Redis, and InMemoryStorage.
Testing is needed for Azure, and Redis

JacobJerrell committed:
bot replies fixed supporting pm_preference

if/elif/else clause to handle user's pm preference options when the bot needs to send a message

kylesykes commented on storage.py:
```diff
@@ -385,7 +395,8 @@ def create_user_entry(self, user_id: str):
             self.POINTS_USED : 0,
-            self.POINTS_RECEIVED : 0
+            self.POINTS_RECEIVED : 0,
+            self.PM_PREFERENCE: 1
```
Does this work with the table schema that exists already @MattPitlyk ?  Do we have to worry about this setting defaulting back to some value each day?

MattPitlyk replied:
This should be fine. The daily resets are for explicitly specified fields, so if we don't put the preference setting in the daily rest function, it won't be reset.

JacobJerrell committed:
Azure and Redis added

Something goofy happened previously. PM Preference options are coded out for Azure and Redis storage options now.
Testing still needed.

JacobJerrell committed:
DS_Store

ignore it

JacobJerrell committed:
small fixes/issue prevention

JacobJerrell committed:
small fixes/issue prevention

MattPitlyk commented on hey_fireball.py:
```diff
@@ -19,7 +19,6 @@
 BOT_ID = os.environ.get("BOT_ID")
 EMOJI = os.environ.get('EMOJI')
 POINTS = os.environ.get('POINTS')
-SELF_POINTS = os.environ.get('SELF_POINTS', "DISALLOW")
```
Why did you remove getting this env var? It's being used in handle_command.

JacobJerrell replied:
I had some issues with github last night. Should be sorted out now

MattPitlyk commented on hey_fireball.py:
```diff
@@ -80,6 +79,9 @@ def __init__(self, msg):
             self.command = self._extract_command()
             self.count = self._extract_count()
+            self.setting = self._extract_setting()
+            self.setting = self._extract_setting() # Find on/off or assume toggle
```
This line is repeated.

JacobJerrell replied:
A few things like this happened last night. I was trying to pick and choose what to push, and made several errors. Sorted it out, and stopped using the computer immediately

JacobJerrell committed:
Merge branch 'develop' of https://github.com/JacobJerrell/hey_fireball into develop

MattPitlyk commented on hey_fireball.py:
```diff
@@ -149,6 +151,29 @@ def _extract_count(self):
+            idx = 2
+            curPref = get_pm_preference(self.requestor_id)
+            try:
+                self.parts[idx]
```
Instead of a try-except, I would check the length of self.parts to determine if which logic branch to execute. Typically, try-except is only using for flow control like this if the statement because caught does something (adds a value to a collection, gets next value in a collection, etc.), but `self.parts[idx]` does not serve a purpose other than to determine if idx=2 is value.

JacobJerrell replied:
try-except has been removed. Now we're checking if self.parts < 3

JacobJerrell committed:
_extract_setting - remove try/except clause

Checking for length of self.parts to decide if it should act as a toggle. Instead of a try/except for an IndexError

JacobJerrell committed:
Removing Local Experiment

Testing the possibility of having the bot reply to user's request for a leaderboard/fullboard.
Bots do not appear to be capable of posting code snippets. So this is an alternative way of 'collapsing' the leaderboard
